scripts/r.mask/r.mask.html

```diff
@@ -4,55 +4,98 @@ <h2>DESCRIPTION</h2>
+raster operations to normal (i.e., all cells of the current region), remove the
+MASK by setting the <b>-r</b> remove MASK flag (<tt>r.mask -r</tt>).
+Alternatively, a mask can be removed using <em>g.remove</em> or by renaming it 
+to any other name with <em>g.rename</em>.
 
 <h2>NOTES</h2>
 
-The above method for specifying a "mask" may seem
-counterintuitive.  Areas inside the MASK are not hidden;
-areas outside the MASK will be ignored until the MASK file
-is removed.
-<p>
-The MASK is read as an integer map. If MASK is actually a
-floating-point map, the values will be converted to integers using the
-map's quantisation rules (this defaults to round-to-nearest, but can
-be changed with r.quant).
+The above method for specifying a "mask" may seem counterintuitive.  Areas 
+inside the MASK are not hidden; areas outside the MASK will be ignored until 
+the MASK file is removed.
 <p>
 <em>r.mask</em> uses <em>r.reclass</em> to create a reclassification of an
-existing raster map and name it <tt>MASK</tt>. A reclass map takes up less space, but
-is affected by any changes to the underlying map from which it was created.
-The user can select category values from the input raster to use in the MASK
-with the <em>maskcats</em> parameter; if <em>r.mask</em> is run from the
+existing raster map and name it <tt>MASK</tt>. A reclass map takes up less 
+space, but is affected by any changes to the underlying map from which it was 
+created. The user can select category values from the input raster to use in the
+MASK with the <em>maskcats</em> parameter; if <em>r.mask</em> is run from the
 command line, the category values listed in <em>maskcats</em> must be quoted
-(see example below).
+(see example below). Note that the <em>maskcats</em> can only be used if the 
+input map is an integer map.
+
+<h3>Different ways to create a MASK</h3>
+
+The <em>r.mask</em> function creates a MASK with values 1 and NULL. But note 
+that a MASK can also be created using other functions that have a raster as 
+output, by naming the output raster 'MASK'. Such layers could have other
+values than 1 and NULL. The user should therefore be aware that grid cells 
+in the MASK map containing <tt>NULL</tt> or <tt>0</tt> will replace data with
+NULL, while cells containing other values will allow data to pass through 
+unaltered. This means that:
 <p>
-Somewhat similar program functions to those performed by
-<em>r.mask</em> can be done using <em>r.mapcalc</em>, 
-<em>g.region</em>, and other commands.
+If a binary map with [0,1] values is used as input in <em>r.mask</em>, all 
+raster cells with 0 and 1 will be part of the MASK. This is because 
+<em>r.mask</em> converts all non-NULL cells to 1.
+
+<div class="code"><pre>
+r.mapcalc -s "map1 = round(rand(0,1))"
+r.mask raster=map1
+</pre></div>
+
+If the same binary map is used as an input in <em>g.copy</em> to create a MASK,  
+only the raster cells with value 1 will be part of the MASK.
```

Comment:
Considering the MASK generated in the above example is still applied, right? Otherwise it's the same case brought up originally.

Reply:
Not sure I understand here. The examples seems ok, but I may be missing something.

Reply:
Hi @veroandreo, not sure either what you mean. Can you elaborate?

Reply:
Well, before the first example, it says "all raster cells with 0 and 1 will be part of the MASK. This is because 
r.mask converts all non-NULL cells to 1." 

Meanwhile, the sentence after the first example seems to say something different and it is not clear which is the "same binary" map it refers to, this is what seems confusing to me. What is being used in the second example is not exactly the same binary map if the mask from the first example is still applied or not... Also, maybe I miss something, but why would  g.copy understand 0 values as NULL to create a mask?

Reply:
Ah... no, this is assuming there is no prior mask in both cases. And it is not g.copy that understands 0 as NULL, it is the raster lib that does for raster layers called MASK. That was in fact were this all started with. The confusion between what r.mask does, and how raster layers with the name MASK are interpreted at lower level.

At the lower level, raster lib interprets a raster layer which is called MASK as "Grid cells in the MASK map containing 0 or NULL will replace data with NULL, while cells containing other values will allow data to pass through unaltered."

So, if you create a raster layer with the name MASK which contains 1 and 0 (e.g., using g.copy, r.mapcalc, .... ) the 1 will be interpreted as part of the MASK, the 0 as NULL.

MASK layers created with r.mask are treated the same way (of course).. the crux is that r.mask reclassifies all values in the input map to 1 and all NULL to NULL (unless you use maskcats). So you only have 1 and NULL in the resulting MASK layers.

Reply:
Thanks for the explanation @ecodiv :) Reading all over again with this explanation in mind, it seems clear... I made a small suggestion as to make it more explicit that the 2 examples are independent